### PR TITLE
README: add backporting instructions

### PR DESCRIPTION
**Pull request by ktdreyer**
Add instructions for how we maintain and release the stable release series.

We'd been following this for a while, so let's get it written down.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -364,6 +364,38 @@ You can now submit a new commit/change that will update the CI system to run a n
 
 It might happen that the CI does not get reloaded so you can simply leave a comment on your PR with "test this please" and it will trigger a new CI build.
 
+# Backporting changes
+
+If a change should be backported to a `stable-*` Git branch:
+
+* Mark your PR with the GitHub label "Backport" so we don't lose track of it.
+* Determine the latest available stable branch:
+  `git branch -r | grep origin/stable`
```
Comment (GregMeno) [inline README.md]: What about this?
git branch -r --list "origin/stable-[0-9].[0-9]" | sort -r | sed 1q

File: README.md [hunk 2]
```diff
@@ -364,6 +364,38 @@ You can now submit a new commit/change that will update the CI system to run a n
...
+git checkout stable-2.2
+git pull --ff-only
+git tag v2.2.5
+git push origin v2.2.5
+```
+
+You can query backport-related items in GitHub:
+* [all PRs marked as backport candidates](https://github.com/ceph/ceph-ansible/labels/backport)
+  to see what work remains
+* [all closed PRs](https://github.com/ceph/ceph-ansible/pulls?q=is%3Apr+is%3Aclosed)
```
Comment (GregMeno) [inline README.md]: I don't think this is quite right.
the open/closed transition isn't capturing https://github.com/ceph/ceph-ansible/pull/1516
it was against master and identified as a backport candidate but then merged and closed

so either backports labeld PRs cannot be closed until backporting is complete
or we'd need another label

I'm for the first but I think that needs to be spelled out

File: README.md [hunk 3]
```diff
@@ -364,6 +364,38 @@ You can now submit a new commit/change that will update the CI system to run a n
...
+
+If a change should be backported to a `stable-*` Git branch:
+
+* Mark your PR with the GitHub label "Backport" so we don't lose track of it.
+* Determine the latest available stable branch:
+  `git branch -r | grep origin/stable`
+* Switch to the stable branch: `git checkout stable-2.2`
+* Cherry-pick your change: `git cherry-pick -x (your-sha1)`
+* Create a new pull request against the `stable-2.2` branch.
+* Ensure that your PR's title has the prefix "backport:", so it's clear
```
Comment (andrewschoen) [inline README.md]: It's also nice to link to the original PR that you're backporting in a comment on the backport PR.

File: README.md (hunk 2 above)
Comment (ktdreyer) [inline README.md]: It is closed when it is merged, so I don't think we can leave it open.
I've updated the queries in my PR here, and added more explanations. Are they ok now?

File: README.md [hunk 4]
```diff
@@ -364,6 +364,38 @@ You can now submit a new commit/change that will update the CI system to run a n
...
 It might happen that the CI does not get reloaded so you can simply leave a comment on your PR with "test this please" and it will trigger a new CI build.
 
+# Backporting changes
+
+If a change should be backported to a `stable-*` Git branch:
+
+* Mark your PR with the GitHub label "Backport" so we don't lose track of it.
+* Determine the latest available stable branch:
+  `git branch -r --list "origin/stable-[0-9].[0-9]" | sort -r | sed 1q`
+* Switch to the stable branch: `git checkout stable-2.2`
```
Comment (andrewschoen) [inline README.md]: I always create a new branch from the stable branch before doing the cherry-pick. If I just cherry-picked into ``stable-2.2`` and then pushed I wouldn't even need to make a PR.

Comment (ktdreyer) [inline README.md]: Whoops, you're right. Amended with this change.

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-update_cluster

File: README.md [hunk 5]
```diff
@@ -364,6 +364,39 @@ You can now submit a new commit/change that will update the CI system to run a n
...
+# Backporting changes
+
+If a change should be backported to a `stable-*` Git branch:
+
+* Mark your PR with the GitHub label "Backport" so we don't lose track of it.
+* Determine the latest available stable branch:
+  `git branch -r --list "origin/stable-[0-9].[0-9]" | sort -r | sed 1q`
+* Create a new local branch for your PR, based on the stable branch:
+  `git checkout -b my-backported-change origin/stable-2.2`
+* Cherry-pick your change: `git cherry-pick -x (your-sha1)`
```
Comment (leseb) [inline README.md]: + `-s` to add a Signed-off-by

File: README.md [hunk 6]
```diff
@@ -364,6 +364,39 @@ You can now submit a new commit/change that will update the CI system to run a n
...
 It might happen that the CI does not get reloaded so you can simply leave a comment on your PR with "test this please" and it will trigger a new CI build.
 
+# Backporting changes
+
+If a change should be backported to a `stable-*` Git branch:
+
+* Mark your PR with the GitHub label "Backport" so we don't lose track of it.
+* Determine the latest available stable branch:
+  `git branch -r --list "origin/stable-[0-9].[0-9]" | sort -r | sed 1q`
+* Create a new local branch for your PR, based on the stable branch:
```
Comment (leseb) [inline README.md]: I'd first run `git checkout stable-2.2 && git pull --rebase origin stable-2.2` and then `git checkout -b my-backported-change`.

File: README.md (hunk 5 above)
Comment (leseb) [inline README.md]: And probably `--allow-empty-message` as well.

Comment (ktdreyer) [inline README.md]: Would you please help me understand why we would want `--allow-empty-message`?

Comment (leseb) [inline README.md]: Forget about that, bad practice.

File: README.md (hunk 6 above)
Comment (andrewschoen) [inline README.md]: Should we mention to do a ``git fetch`` as well?

Comment (leseb) [inline README.md]: Yup, `git fetch origin master`. `git fetch` only will fetch all the remotes, which is not necessary.

Comment (ktdreyer) [inline README.md]: "git fetch" is less typing, right :)

**Pull request merged**